text/0000-sync-networked-vars.md

```diff
@@ -0,0 +1,111 @@
+- Feature Name: `synced_networked_var`
+- Start Date: 2020-Feb-12
+- RFC PR: [Unity-Technologies/com.unity.multiplayer.rfcs#0000](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/0000)
+- Issue: [Unity-Technologies/com.unity.multiplayer#0000](https://github.com/Unity-Technologies/com.unity.multiplayer/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Make the NetworkedVariables synchronized with the tick system.
+
+We suggest two elements:
+- Changing `NetworkedBehaviour::NetworkedBehaviourUpdate()` so that the function runs at most one per Network Tick. When called more than once per Network Tick, it would simply return.
+- NetworkedVariables carrying two pieces of information with their delta updates:
+    - The local tick at which a variable was modified
+    - The remote tick at which the variable was modified, when the host relays the variable to clients, including the source client    
```

Comment:
Would it be possible, to give an option to the end-user to don't send the local & remote ticks, if they don't really need it for a specific NetworkedVar. As I would imagine it will save-up some bandwith?

Reply:
What concerns me is that with the current roadmap, tick system and snapshot are pretty central. If we make the ticks optional, we may end up having to send a boolean telling "is this variable ticked" ? This would result in losing some of the gain bandwidth usage, but also increase complexity.

Reply:
Hey @jeffreyrainy, i totally get what you mean, but I was thinking that maybe those variables are set to '-1' when they are not sent, and that would be similar to the flag that you mentioned, we can just check against client-side without the having to introduce another variable to server that purpose.

Reply:
Yeah. I get you. Let's keep that in mind when we implement it. I'll check which gain we can have around making them optional